File: code/animal_accouting_system/src/main/java/geekbrains/animals/Model/Animal/Pet/Impl/Dog.java
package geekbrains.animals.Model.Animal.Pet.Impl;


import geekbrains.animals.Model.Animal.Interfaces.Barkable;
import geekbrains.animals.Model.Animal.Pet.Pet;

import java.time.LocalDate;

public class Dog extends Pet implements Barkable {

    public Dog(String name, LocalDate birthdate) {
        super(name, birthdate);
        setType("собака");
    }

    @Override
    public void bark() {
        System.out.println("Woof!");
    }
}

File: code/animal_accouting_system_golang/src/animal/pet/dog/dog.go
package dog

import (
	"aac/src/animal/pet"
	"time"
)

type Dog struct {
	pet.Pet
}

func NewDog(name string, birthDate time.Time) Dog {
	return Dog{
		Pet: pet.Pet{
			Name:      name,
			BirthDate: birthDate,
			Type:      "Dog",
		},
	}
}
